bundles/framework/hierarchical-layerlist/service/LayerlistExtenderService.js

```diff
@@ -0,0 +1,424 @@
+             *                                     cls: 'active-cls'
+             *                                 }
+             */
+            addSubgroupSubgroupTool: function(id, handler, options) {
+                if (this._subgroupSubgroupTools[id]) {
+                    _log.warn('Subgroup subgroup tool "' + id + '" allready defined.');
+                    return;
+                }
+                if (typeof handler !== 'function') {
+                    _log.warn('Subgroup subgroup tool "' + id + '" has no any handler, not added.');
+                    return;
+                }
+
+                this._subgroupSubgroupTools[id] = {
+                    handler: handler,
+                    options: options
+                };
+
+                this.trigger('subgroupsubgrouptool.added', {
+                    id: id,
+                    handler: handler,
+                    options: options
+                });
+            },
+
+            /**
+             * Get layer tool.
+             * @method getLayerTool
+             * @param  {String}    id if defined return tool id corresponding tool, if not defeined return all layer tools.
+             * @return {Object}    wanted layer tool or all layer tools
+             */
+            getLayerTool: function(id) {
+                if (!id) {
+                    return this._layerTools;
+                }
+                return this._layerTools[id];
+            },
+            /**
+             * Add layer tool
+             * @method addLayerTool
+             * @param  {String}     id      layer tool id
+             * @param  {Function}   handler layer tool handler
+             * @param  {Object}     options layer tool options:
+             *                                 {
+             *                                     cls: 'active-cls'
+             *                                 }
+             */
+            addLayerTool: function(id, handler, options) {
+                if (this._layerTools[id]) {
+                    _log.warn('Layer tool "' + id + '" allready defined.');
+                    return;
+                }
+                if (typeof handler !== 'function') {
+                    _log.warn('Layer tool "' + id + '" has no any handler, not added.');
+                    return;
+                }
+
+                this._layerTools[id] = {
+                    handler: handler,
+                    options: options
+                };
+                this.trigger('layertool.added', {
+                    id: id,
+                    handler: handler,
+                    options: options
+                });
+            },
+            /**
+             * Sets admin role
+             * @method setAdmin
+             * @param  {Boolean} isAdmin is admin
+             */
+            setAdmin: function(isAdmin) {
+                this._hasAdmin = isAdmin;
+                this.trigger('admin.changed', this._hasAdmin);
```

Comment:
Why does this need to be tracked separately? Why not just react to layer tool changes when admin bundle adds them?

Reply:
Only thing what use this is at when user is admin then shows also empty groups in layers so admin can add edit/delete group and also add layers.

Reply:
Maybe rename this for clarity? Show/IncludeEmptyGroups() or add a comment